java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll

```diff
@@ -0,0 +1,76 @@
+/** Provides classes and predicates for exclusions related to MaD models. */
+
+import java
+
+pragma[nomagic]
+private predicate isTestPackage(Package p) {
+  p.getName()
+      .matches([
+          "org.junit%", "junit.%", "org.mockito%", "org.assertj%",
+          "com.github.tomakehurst.wiremock%", "org.hamcrest%", "org.springframework.test.%",
+          "org.springframework.mock.%", "org.springframework.boot.test.%", "reactor.test%",
+          "org.xmlunit%", "org.testcontainers.%", "org.opentest4j%", "org.mockserver%",
+          "org.powermock%", "org.skyscreamer.jsonassert%", "org.rnorth.visibleassertions",
+          "org.openqa.selenium%", "com.gargoylesoftware.htmlunit%", "org.jboss.arquillian.testng%",
+          "org.testng%", "%.test%"
+        ])
+}
+
+/**
+ * A test library.
+ */
+private class TestLibrary extends RefType {
+  TestLibrary() { isTestPackage(this.getPackage()) }
+}
+
+private predicate isInTestFile(File file) {
+  file.getAbsolutePath().matches("%src/test/%") or
+  file.getAbsolutePath().matches("%/guava-tests/%") or
+  file.getAbsolutePath().matches("%/guava-testlib/%")
+}
+
+private predicate isJdkInternal(CompilationUnit cu) {
```

Comment:
Would it make sense to refactor [this code](https://github.com/github/codeql/blob/d94ed1b4a31243f9050902b1a3c7c2a7cbd546a6/java/ql/src/utils/modelgenerator/internal/CaptureModelsSpecific.qll#L35) to use the predicate above?

Reply:
Yes, that's part of what I'm planning to refactor in the TODO that I mentioned in the description above. 🙂 
`TODO: Refactors DataFlowTargetApi/TargetApiSpecific and ExternalApi to use the new ModelExclusions.qll file.`